src/gui/EntryPreviewWidget.cpp

```diff
@@ -240,13 +240,19 @@ void EntryPreviewWidget::updateEntryGeneralTab()
         m_ui->togglePasswordButton->setVisible(false);
     }
 
-    if (config()->get(Config::Security_HideNotes).toBool()) {
-        setEntryNotesVisible(false);
-        m_ui->toggleEntryNotesButton->setVisible(!m_ui->entryNotesTextEdit->toPlainText().isEmpty());
-        m_ui->toggleEntryNotesButton->setChecked(false);
-    } else {
-        setEntryNotesVisible(true);
+    if (m_currentEntry->notes().isEmpty()) {
+        m_ui->entryNotesTextEdit->setVisible(false);
         m_ui->toggleEntryNotesButton->setVisible(false);
+    } else {
+        m_ui->entryNotesTextEdit->setVisible(true);
+        if (config()->get(Config::Security_HideNotes).toBool()) {
+            setEntryNotesVisible(false);
+            m_ui->toggleEntryNotesButton->setVisible(!m_ui->entryNotesTextEdit->toPlainText().isEmpty());
+            m_ui->toggleEntryNotesButton->setChecked(false);
+        } else {
+            setEntryNotesVisible(true);
+            m_ui->toggleEntryNotesButton->setVisible(false);
+        }
     }
```

Comment:
I'm not familiar with the rest of KPXC's code, nor with Qt yet, but just taking these lines on their own, this looks messy. A bunch of branching code setting multiple inter-dependent state in a complex manner. This screams "bugs!" at me.

1. First, why not actually clear the notes member in the model by default, when it's initialized? Or *always* load the notes from the entry, so that here it would get overwritten by an empty string? That's a more sensible solution to me than this UI hack.
1. Related to that, having two different ways to check if the notes are empty (lines 243 and 250) feels wrong. I understand these check slightly different things, but if the above point is fixed, then the 2nd form (line 250) should work for both (and can be checked once).
1. From line 250, `m_ui->entryNotesTextEdit`, it appears this can be editable in some cases? If so, then you shouldn't be hiding the edit box just because it's initially empty (line 244). Or, if it's never editable here, then why is this an edit box and not a label?
1. (For that matter, if it's editable, why are the "edit" and "view" views not separated to begin with? Though I realize that separating them is beyond the scope of this PR.)
1. Ignoring the above, if you go ahead with this UI approach, I would at least rearrange the code to reduce the branching complexity. How about this:
    1. Check the two conditions in advance, and put their results in boolean variables, `isNotesEmpty` and `isHideNotes`.
    1. Set each boolean state exactly once, passing the correct combination of conditions to the boolean parameter. For example, `m_ui->toggleEntryNotesButton->setVisible(isHideNotes && !isNotesEmpty);`. This would clearly spell out what the compound condition is for each state.
1. Finally, this could really use some unit tests, if possible (if there already are, then at least for the new bug scenario). And ideally they should be written *before* changing the actual code.

I realize @droidmonkey has already approved this PR, and it's not my place to overrule, but these are just some points to consider.

Reply:
There are a lot of quirks with this dialog to make it work and function the way it does. It's not clean, welcome to gui programming.

Reply:
Then how about point 5, at least? I think that would be cleaner still.

Reply:
Reasonable yes